aim/common/utils.py

```diff
@@ -58,6 +60,20 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+# created a common function for tackling issues of sorting a list of dict
+# or list of string for Py3 support
+def sorted_if_list(attr):
+    if six.PY3:
+        if isinstance(attr, list):
+            if attr and isinstance(attr[0], dict):
+                return sorted(attr, key=lambda i: list(i.keys())[0])
```

Comment:
I agree my code comment was misleading.

Here we can't use your solution because:

`sorted()` on dict returns list of 'keys' of that dict in Py2.
To replicate that behaviour in Py3, we will need to use `sorted(attr, key=lambda i: list(i.keys())[0])`

https://docs.python.org/2.7/howto/sorting.html

Reply:
```
abcd = [{'name': {'abcd':'2'}, 'path': 'c'}, {'name': {'abcd':'1'}, 'path': 'd'}]
print(sorted(abcd))
```
If we run this in Python3: 
1) via `sorted(attr, key=lambda i: list(i.keys())[0])`
2) via `sorted(attr, key=lambda i: sorted(i.items()))`

Here, (1) will run but won't give correct sorted nested dict but since we are comparing 2 list of dicts with same function it is not creating an error, and (2) will give error `'<' not supported between instances of 'dict' and 'dict'`

[1. vs 2. repl check](https://repl.it/repls/WoozyTautSign)

Meanwhile I am thinking of a solution for this.

Reply:
Sure I'm open for other possible solution here. But did it fail any UTs if you use "sorted(attr, key=lambda i: sorted(i.items()))"? If not then we are probably okay because it means its enough to cover all the use cases in AIM. The comment has to be adjusted accordingly also depends on what's the final solution here.

Reply:
As I have mentioned earlier, `sorted(attr, key=lambda i: sorted(i.items()))` will face issues when we have list of nested dict.
Actually it is failing because we have list of nested dicts  in our tests. I will give you an example from the error which I got from failed testcase `aim.tests.unit.agent.aid_universes.test_aci_universe.TestAciUniverse.test_get_resources` using `sorted(attr, key=lambda i: sorted(i.items()))` :

`[{'faultInst': {'attributes': {'dn': 'uni/tn-t1/ap-a1/epg-test/fault-951', 'descr': 'cannot resolve', 'code': '951', 'severity': 'warning', 'cause': 'resolution-failed'}}}, {'fvBD': {'attributes': {'arpFlood': 'no', 'dn': 'uni/tn-test-tenant/BD-test', 'epMoveDetectMode': '', 'ipLearning': 'yes', 'limitIpLearnToSubnets': 'no', 'unicastRoute': 'yes', 'unkMacUcastAct': 'proxy', 'nameAlias': ''}}}, {'fvRsCtx': {'attributes': {'dn': 'uni/tn-test-tenant/BD-test/rsctx', 'tnFvCtxName': ''}}}, {'vzSubj': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s', 'nameAlias': ''}}}, {'vzInTerm': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s/intmnl'}}}, {'vzOutTerm': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s/outtmnl'}}}, {'vzRsSubjFiltAtt': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s/rssubjFiltAtt-f', 'tnVzFilterName': 'f'}}}, {'vzRsFiltAtt': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s/intmnl/rsfiltAtt-g', 'tnVzFilterName': 'g'}}}, {'vzRsFiltAtt': {'attributes': {'dn': 'uni/tn-t1/brc-c/subj-s/outtmnl/rsfiltAtt-h', 'tnVzFilterName': 'h'}}}]`

Error is: `TypeError: '<' not supported between instances of 'dict' and 'dict'`

PS. There's a library which can be used to compare dicts, list, strings, etc. which can 
https://pypi.org/project/deepdiff/